src/main/java/executor/command/CommandHelp.java

```diff
@@ -1,32 +1,51 @@
 package executor.command;
 
 import executor.task.TaskList;
+import interpreter.Parser;
 import ui.Ui;
 import ui.Wallet;
 
+import java.security.spec.ECField;
+import java.util.ArrayList;
+import java.util.Collections;
+
 public class CommandHelp extends Command {
 
     /**
      * Constructor to provide the user with the details about the commands available.
      */
     public CommandHelp(String userInput) {
         this.userInput = userInput;
-        this.description = "Provides the user with all the available commands and descriptions.";
+        this.description = "Provides the user with all the available commands and descriptions.\n"
+                + "FORMAT : help \n";
         this.commandType = CommandType.HELP;
     }
 
     @Override
     public void execute(TaskList taskList) {
-        for (String s : CommandType.getNames()) {
-            if (!s.equals("ERROR") && !s.equals("TASK") && !s.equals("BLANK")) {
-                CommandType commandType = CommandType.valueOf(s);
-                Command c = Executor.createCommand(commandType, "null");
-                String commandDesc = c.getDescription();
-
-                System.out.println(s.toUpperCase() + " - " + commandDesc);
-            }
+        String checkerForExtraInput;
+        checkerForExtraInput = Parser.parseForPrimaryInput(CommandType.HELP, userInput);
+
+        if (!checkerForExtraInput.isEmpty()) {
```

Comment:
I would imagine it would be better to allow for help to take in a primaryInput so that you can look for that specific command and if it doesn't exist, just print command doesn't exist. In general, Help commands usually allows ppl to specify what Command their looking for so do you mind implementing that?

Reply:
I have implemented it in the recent commit.